### PR TITLE
[ELY-224] Add ClearPassword  support to SimpleMapBackedSecurityRealm.

### PR DESCRIPTION
**Pull request by darranl**

### CODE REVIEW

File: src/main/java/org/wildfly/security/auth/provider/SimpleMapBackedSecurityRealm.java [hunk 1]
```diff
@@ -157,21 +159,25 @@ public AuthorizationIdentity getAuthorizationIdentity() {
...
         }
 
+        @Override
         public boolean verifyCredential(final Object credential) throws RealmUnavailableException {
-            if (credential instanceof char[]) try {
-                final SimpleRealmEntry entry = map.get(name);
-                if (entry == null) {
+            try {
+                if (credential instanceof char[]) {
+                    final Password password = map.get(name).getPassword();
```
Comment (dmlloyd) [inline src/main/java/org/wildfly/security/auth/provider/SimpleMapBackedSecurityRealm.java]: NPE if the map doesn't contain the name; IMO the local var extraction/null check should remain.

File: src/main/java/org/wildfly/security/auth/provider/SimpleMapBackedSecurityRealm.java [hunk 2]
```diff
@@ -157,21 +159,29 @@ public AuthorizationIdentity getAuthorizationIdentity() {
...
-                final SimpleRealmEntry entry = map.get(name);
+            try {
+                SimpleRealmEntry entry = map.get(name);
                 if (entry == null) {
                     return false;
                 }
-                final Password password = entry.getPassword();
-                return PasswordFactory.getInstance(password.getAlgorithm()).verify(password, (char[]) credential);
+
+                if (credential instanceof char[]) {
```
Comment (pedroigor) [inline src/main/java/org/wildfly/security/auth/provider/SimpleMapBackedSecurityRealm.java]: Should we also expect and accept String as credential ? 

Also, I've noticed that there are ```SecurityRealm``` implementations that only accept ```ClearPassword```. Should we change them as well ?

Comment (darranl) [inline src/main/java/org/wildfly/security/auth/provider/SimpleMapBackedSecurityRealm.java]: IMO we should eliminate the use of String for credential validation - generally it is a bad practice as we can not clear it's contents.

Comment (pedroigor) [inline src/main/java/org/wildfly/security/auth/provider/SimpleMapBackedSecurityRealm.java]: Ok, so I'm going to remove from the realms I'm implementing.

**Pull request merged**